src/posture_estimator.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file posture_estimator.h
+ * @brief Interface for representation of user's pose
+ *
+ * @copyright Copyright (C) 2021  Conor Begley
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "posture_estimator.h"
+
+#include <stdio.h>
+
+#include <cmath>
+
+namespace PostureEstimating {
+
+Pose createPose() {
+  Pose p;
+
+  for (int i = JointMin; i <= JointMax; i++) {
+    p.joints[i] =
+        reinterpret_cast<ConnectedJoint *>(malloc(sizeof(ConnectedJoint)));
+    p.joints[i]->joint = static_cast<Joint>(i);
+    p.joints[i]->coord = {0, 0, PostProcessing::Untrustworthy};
+    p.joints[i]->upper_angle = 0;
+    p.joints[i]->lower_angle = 0;
+  }
+
+  p.joints[JointMin]->upper_connected_joint = nullptr;
+  p.joints[JointMin]->lower_connected_joint = p.joints[JointMin + 1];
+
+  for (int i = JointMin + 1; i <= JointMax; i++) {
+    p.joints[i]->upper_connected_joint = p.joints[i - 1];
+    p.joints[i]->lower_connected_joint = p.joints[i + 1];
+  }
+
+  p.joints[JointMax]->upper_connected_joint = p.joints[JointMax - 1];
+  p.joints[JointMax]->lower_connected_joint = nullptr;
```

Comment:
If I understand correctly, we are effectively building a doubly-linked list here and also storing it in an array. To my mind that means we are storing two extra pointers per joint than we need. I'm sure there is a good reason for this design choice, but I just wanted to double check.

Reply:
We discussed this before, and I asked about it and I think you said that is was a small overhead and there maybe an advantage?

Reply:
Ah OK, I only vaguely remember that conversation, sorry. As we'd said, any overhead will probably be small and I don't want to prevent this being merged as it isn't a critical thing (I think).

Reply:
But I'm also happy to remove

Reply:
We can also discuss on the daily stand up